backend/service/topology/cache.go

```diff
@@ -99,64 +100,110 @@ func (c *client) startTopologyCache(ctx context.Context) {
 	<-ctx.Done()
 }
 
+// Processes all topology cache update requests.
+// This supports insert batching which defaults to 1, and can be configured by settting BatchInsertSize.
+// Notably a flush will happen every 30 seconds, ensuring all items make it into cache even if the
+// configured BatchInsertSize is not met.
 func (c *client) processTopologyObjectChannel(ctx context.Context, objs <-chan *topologyv1.UpdateCacheRequest, service string) {
-	for obj := range objs {
-		c.scope.Tagged(map[string]string{
-			"service": service,
-		}).SubScope("cache").Gauge("object_channel.queue_depth").Update(float64(len(objs)))
-
-		switch obj.Action {
-		case topologyv1.UpdateCacheRequest_CREATE_OR_UPDATE:
-			if err := c.setCache(ctx, obj.Resource); err != nil {
-				c.log.Error("Error setting cache", zap.Error(err))
+	var batchInsert []*topologyv1.Resource
+
+	for {
+		select {
+		case obj, ok := <-objs:
+			if !ok {
+				return
+			}
+
+			c.scope.Tagged(map[string]string{
+				"service": service,
+			}).SubScope("cache").Gauge("object_channel.queue_depth").Update(float64(len(objs)))
+
+			switch obj.Action {
+			case topologyv1.UpdateCacheRequest_CREATE_OR_UPDATE:
+				batchInsert = append(batchInsert, obj.Resource)
+			case topologyv1.UpdateCacheRequest_DELETE:
+				if err := c.deleteCache(ctx, obj.Resource.Id, obj.Resource.Pb.TypeUrl); err != nil {
+					c.log.Error("Error deleting cache", zap.Error(err))
+				}
+			default:
+				c.log.Warn("UpdateCacheRequest action is not implemented", zap.String("action", obj.Action.String()))
 			}
-		case topologyv1.UpdateCacheRequest_DELETE:
-			if err := c.deleteCache(ctx, obj.Resource.Id, obj.Resource.Pb.TypeUrl); err != nil {
-				c.log.Error("Error deleting cache", zap.Error(err))
+
+			if len(batchInsert) >= int(c.config.Cache.BatchInsertSize) {
```

Comment:
nit: store int(c.config.Cache.BatchInsertSize) as a member of the struct rather than reading it and doing a type conversion every time.

Reply:
I really wish I could use that config as the source of truth and not spill these fields over into the struct.